packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,233 @@
+
+  const copyValue = () => {
+    navigator.clipboard.writeText(value.toString())
+    setOpenCopyAlert(true)
+  }
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems={alignment} gap={Spacing.xs}>
+      <Typography variant="bodyXsRegular" color="textTertiary">
+        {label}
+        {tooltip && (
+          <Tooltip arrow placement="top" title={tooltip}>
+            <span>
+              &nbsp;
+              <InfoOutlinedIcon sx={{ fontSize: '1.25em' }} />
+            </span>
+          </Tooltip>
+        )}
+      </Typography>
+
+      {loading ? (
+        <Skeleton
+          variant="text"
+          width="100%"
+          sx={{
+            fontSize: TYPOGRAPHY_VARIANTS[MetricSize[size]],
+          }}
+        />
+      ) : (
+        <Box display="flex" gap={Spacing.xs} alignItems="baseline">
+          <Tooltip
+            arrow
+            placement="bottom"
+            title={value.toLocaleString()}
+            onClick={copyValue}
+            sx={{
+              cursor: 'pointer',
+            }}
+          >
+            <Box display="flex" gap={Spacing.xxs} alignItems="baseline">
+              {unit?.position === 'prefix' && (
+                <Typography variant={MetricSize[size]} color="textSecondary">
+                  {unit.symbol}
+                </Typography>
+              )}
+
+              <Typography variant={MetricSize[size]} color="textPrimary">
+                {formatter(abbreviate ? abbreviateNumber(value) : value)}
+              </Typography>
+
+              {abbreviate && (
+                <Typography variant={MetricSize[size]} color="textPrimary" textTransform="capitalize">
+                  {scaleSuffix(value)}
+                </Typography>
+              )}
+
+              {unit?.position === 'suffix' && (
+                <Typography variant={MetricSize[size]} color="textSecondary">
+                  {unit.symbol}
+                </Typography>
+              )}
+            </Box>
+          </Tooltip>
+
+          {(change || change === 0) && (
+            <Typography
+              variant="highlightM"
+              color={change > 0 ? 'success' : change < 0 ? 'error' : 'textHighlight'}
+              sx={{ marginInlineStart: Spacing.xs }} // Not part of Figma, but my own addition
+            >
+              {formatChange(change)}%
+            </Typography>
+          )}
+        </Box>
+      )}
+
+      {notional !== undefined && (
+        <Typography variant="highlightXsNotional" color="textTertiary">
+          {notionalUnit?.position === 'prefix' && notionalUnit.symbol}
+          {notionalFormatter(notionalAbbreviate ? abbreviateNumber(notional) : notional)}
+          {notionalAbbreviate && scaleSuffix(notional)}
+          {notionalUnit?.position === 'suffix' && notionalUnit.symbol}
+        </Typography>
+      )}
+
+      <Snackbar open={openCopyAlert} autoHideDuration={6000} onClose={() => setOpenCopyAlert(false)}>
```

Comment:
the `autoHideDuration` shouldn't be hardcoded. Let's either use a constant or configure this in `components.ts` like `muiSnackbar: {defaultProps: {autoHideDuration: 6000}}`

Reply:
I've decided to create a const variable on top called `COPY_ALERT_DURATION`. I don't like the idea of setting it default to 6000ms for every snackbar, as we might also use it in the future for more important web3 transaction related alerts which you don't want to hide away as quick.

Was thinking about putting it in `0_primitives.ts` but wasn't sure if that's the right place, so just put it here as it's only related to this specific component for now.